packages/reaction-core/client/templates/products/productGrid/item/item.js

```diff
@@ -104,31 +104,28 @@ Template.productGridItems.helpers({
 Template.productGridItems.events({
   "click [data-event-action=productClick]": function (event) {
     if (ReactionCore.hasPermission("createProduct")) {
-      if (event.shiftKey) {
+      if (event.metaKey || event.ctrlKey || event.shiftKey) {
         event.preventDefault();
-
         let checkbox = $(`input[type=checkbox][value=${this._id}]`);
-        let checked = checkbox.prop("checked");
-
-        checkbox
-          .prop("checked", !checked)
-          .trigger("change");
+        let selected = $('li.product-grid-item.active').length;
+        if (event.shiftKey && selected > 0) {
+          let indexes = [$('li.product-grid-item').index(checkbox.parents('li.product-grid-item')),
+            $('li.product-grid-item').index($('li.product-grid-item.active').get(0)),
+            $('li.product-grid-item').index($('li.product-grid-item.active').get(selected-1))];
+          for (let i=_.min(indexes); i<=_.max(indexes); i++) {
+            $('input[type=checkbox]', $('li.product-grid-item').get(i)).prop("checked", true).trigger("change");
+          }
+        } else {
+          checkbox.prop("checked", !checkbox.prop("checked")).trigger("change");
+        }
       }
```

Comment:
So these are small details, but there are a few best practice tweaks that should probably be made here.  First, there are several jQuery calls on the same elements... `$('li.product-grid-item')`, etc. which should be cached in a variable and reused. Second, all of those DOM queries can be scoped to the template by passing it into the event handler.  And finally, variables that aren't expected to change should be defined with `const` instead of `let` (This one wasn't really on you. I realize there were several `let` declarations there to begin with.  Just trying to clean up!)

So anyway, the event handler should look something like...

```js
"click [data-event-action=productClick]" (event, tmpl) {
  if (ReactionCore.hasPermission("createProduct")) {
    if (event.metaKey || event.ctrlKey || event.shiftKey) {
      event.preventDefault();
      
      const $items = tmpl.$('li.product-grid-item');
      const $activeItems = tmpl.$('li.product-grid-item.active');
      const $checkbox = tmpl.$(`input[type=checkbox][value=${this._id}]`);
      const selected = $activeItems.length;
      
      if (event.shiftKey && selected > 0) {
        const indexes = [
          $items.index($checkbox.parents('li.product-grid-item')),
          $items.index($activeItems.get(0)),
          $items.index($activeItems.get(selected-1))
        ];
        for (let i=_.min(indexes); i<=_.max(indexes); i++) {
          tmpl.$('input[type=checkbox]', $items.get(i)).prop("checked", true).trigger("change");
        }
      } else {
        $checkbox.prop("checked", !$checkbox.prop("checked")).trigger("change");
      }
    }
  }
}
```